### PR TITLE
Color source components and surface reflections

### PR DESCRIPTION
**Pull request by keptsecret**
Ported over color source component from maxwell's fork  
Adding SSR with skybox fallback for surface reflections  
--> requires detailed metallic and roughness maps to look good, maybe add that

Currently limiting color sources to 5 per level

### CODE REVIEW

**Comment by keptsecret** [comment]
Now starts in full screen and uses monitor screen resolution.  
Needs taking a look at, testing with different materials (maybe varied surfaces), and some tweaking for SSR but is mostly good to go.

Current material default roughness=0.5, metallic=0, as shown. Last picture is with roughness=0.8 and tones reflections down more. Think I might tune reflection strength curve a bit more.

Screenshot 2023-09-28 180351
Screenshot 2023-09-28 180507
Screenshot 2023-09-28 180733

File: data/shaders/postprocess.frag [hunk 1]
```diff
@@ -47,6 +47,19 @@ vec3 aces_fitted(vec3 color) {
...
 
+const float A = 0.15;
+const float B = 0.50;
+const float C = 0.10;
+const float D = 0.20;
+const float E = 0.02;
+const float F = 0.30;
+const float W = 11.2;
+
+vec3 Uncharted2(vec3 color) {
```
Comment (kevinsadi) [inline data/shaders/postprocess.frag:58]: huh

File: src/engine/scenes/TestSceneKevin.cpp [hunk 2]
```diff
@@ -79,10 +80,16 @@ namespace lei3d
 		physicsPlaygroundCollider->Init();
 		physicsPlaygroundCollider->SetColliderToModel(*playgroundModel);
 
+		// Test color source
```
Comment (kevinsadi) [inline src/engine/scenes/TestSceneKevin.cpp:93]: My only question is that I'm having a hard time seeing the color sphere in effect, where is it right now with reference to the scene and what am I supposed to see?

**Review by kevinsadi — APPROVED** [review]
My main question before this gets merged in is about the color sources, I'm not really seeing their effect, maybe I'm not looking in the right places?

File: src/engine/scenes/TestSceneKevin.cpp (hunk 2 above)
Comment (kevinsadi) [reply src/engine/scenes/TestSceneKevin.cpp:93]: Oh psych, I see it, the red in the back of the starting platform

Comment (kevinsadi) [reply src/engine/scenes/TestSceneKevin.cpp:93]: okay nvm I see

File: data/shaders/postprocess.frag (hunk 1 above)
Comment (kevinsadi) [reply data/shaders/postprocess.frag:58]: Is this like Uncharted2's tone mapping?

Comment (keptsecret) [reply data/shaders/postprocess.frag:58]: Yeah, I was playing around with ones that made the scene not too dark. ACES is pretty dark, will need exposure as well. This one with exposure factor 16 makes it quite bright. You can play around and see which you'd prefer.

Comment (kevinsadi) [reply data/shaders/postprocess.frag:58]: bet

**Pull request merged**